### PR TITLE
api exercises

### PR DESCRIPTION
**Pull request by dpenny**
## What
skeleton Models+ serializers,  some initial APIs

## Looking for feedback on
-are functions defined in the right place
-general architecture
-OrderItem as the model that has Item and Quantity? is there a better way? not sure how I feel about it being many to many

## Todos
-beef out models and serializers with more functionality 

@marknach

### CODE REVIEW

File: ddl/api.py [hunk 1]
```diff
@@ -1,33 +1,64 @@
...
+    oc = OrderCart.objects.get(id=order_cart_id)
+    serializer = OrderCartSerializer(customer)
+    return Response(serializer.data)
+
+  # Check out an order cart
+  # is this the best place to put the function?
+  def checkout_order_cart(self, order_cart_id, request,  *args, **kwargs):
+     oc = OrderCart.objects.get(id=order_cart_id)
+     oc.has_checked_out = True
+     oc.save()
```
Comment (marknach) [inline ddl/api.py]: we could decouple this API from the model implementation by creating a service module 
something like `/order_cart/checkout.py` that contains a function w/ this logic

File: ddl/api.py [hunk 2]
```diff
@@ -1,33 +1,64 @@
...
+     # do other checkout logic here
+    return True
```
Comment (marknach) [inline ddl/api.py]: also want to return a response here

File: ddl/models.py [hunk 3]
```diff
@@ -47,20 +55,43 @@ class Meta:
...
   class Meta:
     db_table = 'order_cart'
 
+  #future fields: promo code, etc
   order_items = models.ManyToManyField(OrderItem, blank=True)
 
+
+  #Add food items to the order cart
+  #sketch???
+  def update(self, order_item_list):
```
Comment (marknach) [inline ddl/models.py]: this `update` seems more like an `add_order_item_to_cart`

Comment (marknach) [reply ddl/models.py]: we could also pull this out of the model & pass the instance in as a param

File: ddl/models.py [hunk 4]
```diff
@@ -47,20 +55,43 @@ class Meta:
...
+    self.order_items.append(order_item_list)
+    self.save()
+
+  #Create a new ordercart
+  @classmethod
+  def create(cls, order_items):
```
Comment (marknach) [inline ddl/models.py]: What does this do vs just `OrderCart.objects.create()`?

File: ddl/models.py [hunk 5]
```diff
@@ -47,20 +55,43 @@ class Meta:
...
     db_table = 'order_table'
 
   customer = models.ForeignKey(Customer)
+  restaurant = models.ForeignKey(Restaurant)
   order_cart = models.ForeignKey(OrderCart)
-  #add delivery once added to table
+  address = models.ForeignKey(Address)
+  has_checked_out = models.BooleanField()
 
+  def update_address(self, address):
```
Comment (marknach) [inline ddl/models.py]: this method seems unnecessary at the current stage, but if so, I would pull out of model as well

File: ddl/models.py (hunk 3 above)
Comment (dpenny) [reply ddl/models.py]: 👍 And put it in in OrderCartDetailView?

Comment (dpenny) [reply ddl/models.py]: or also better to decouple from view?

Comment (marknach) [reply ddl/models.py]: checkout this: https://github.com/doordash/doordash-backend-standards/blob/master/business-logic.md#crud-organization

**Pull request merged**